railgun/internal/ctrlutils/ingress-status.go

```diff
@@ -124,83 +130,109 @@ func toKnativeLBStatus(coreLBStatus []apiv1.LoadBalancerIngress) []knative.LoadB
-	namespace, name, err := retrieveNSAndNM(svc)
-	if err != nil {
-		return fmt.Errorf("failed to generate UDP client. err %v", err)
-	}
-
-	ingCli := cli.NetworkingV1().Ingresses(namespace)
-	cli.NetworkingV1beta1()
-	if err != nil {
-		return fmt.Errorf("failed to generate UDP client. err %v", err)
-	}
-
-	retry := 0
-	for retry < statusUpdateRetry {
-		curIng, err := ingCli.Get(ctx, name, metav1.GetOptions{})
-		if err != nil || curIng == nil {
-			log.Errorf("failed to fetch Ingress %v/%v: %v. retrying...", namespace, name, err)
-			retry++
-			time.Sleep(time.Second)
-			continue
-		}
+	for _, route := range svc.Routes {
+		routeInf := strings.Split(*((*route).Name), ".")
+		namespace := routeInf[0]
+		name := routeInf[1]
+		log.Infof("route name %s namespace %s", name, namespace)
+
+		ingCli := cli.NetworkingV1().Ingresses(namespace)
+		retry := 0
+		for retry < statusUpdateRetry {
+			curIng, err := ingCli.Get(ctx, name, metav1.GetOptions{})
+			if err != nil || curIng == nil {
+				log.Errorf("failed to fetch Ingress %v/%v: %v. retrying...", namespace, name, err)
+				retry++
+				time.Sleep(time.Second)
+				continue
+			}
 
-		var status []apiv1.LoadBalancerIngress
-		sort.SliceStable(status, lessLoadBalancerIngress(status))
-		curIPs := curIng.Status.LoadBalancer.Ingress
+			var status []apiv1.LoadBalancerIngress
+			sort.SliceStable(status, lessLoadBalancerIngress(status))
+			curIPs := curIng.Status.LoadBalancer.Ingress
 
-		status = SliceToStatus(ips)
-		if ingressSliceEqual(status, curIPs) {
-			log.Debugf("no change in status, update ingress v1 skipped")
-			return nil
-		}
+			status = SliceToStatus(ips)
+			if ingressSliceEqual(status, curIPs) {
+				log.Debugf("no change in status, update ingress v1 skipped")
+				return nil
+			}
 
-		curIng.Status.LoadBalancer.Ingress = status
+			curIng.Status.LoadBalancer.Ingress = status
 
-		_, err = ingCli.UpdateStatus(ctx, curIng, metav1.UpdateOptions{})
-		if err == nil {
-			break
+			_, err = ingCli.UpdateStatus(ctx, curIng, metav1.UpdateOptions{})
+			if err == nil {
+				break
+			}
+			log.Errorf("failed to update Ingress V1 status. %v. retrying...", err)
+			time.Sleep(time.Second)
+			retry++
 		}
-		log.Errorf("failed to update Ingress V1 status. %v. retrying...", err)
-		time.Sleep(time.Second)
-		retry++
 	}
 
-	return fmt.Errorf("ingress_status successfully updated UDPIngress status")
-
+	log.Info("successfully updated networkingv1 Ingress status")
+	return nil
 }
 
 // UpdateUDPIngress update udp ingress status
 func UpdateUDPIngress(ctx context.Context, logger logr.Logger, svc file.FService, kiccli *kicclientset.Clientset,
 	ips []string) error {
-	namespace, name, err := retrieveNSAndNM(svc)
-	if err != nil {
-		return fmt.Errorf("failed to generate UDP client. err %v", err)
+	for _, route := range svc.Routes {
+		routeInf := strings.Split(*((*route).Name), ".")
+		namespace := routeInf[0]
+		name := routeInf[1]
+		log.Infof("route name %s namespace %s", name, namespace)
```

Comment:
similar suggestion to above
```suggestion
		log.Infof("updating status for v1beta1.UDPIngress route: name %s namespace %s", name, namespace)
```